src/filepattern/cpp/internal/vectorpattern.cpp

```diff
@@ -31,14 +32,16 @@ void VectorPattern::matchFiles(){
         file = VectorParser::getFileName(line); // get filename from line of stitching vector
         if(regex_match(file, sm, this->getRegexExpression())){ // match to regex. groups are in sm
             temp = getVariableMap(file, sm); // get the variable map for the filename
-            VectorParser::parseVectorLine(temp, line, this->STITCH_VARIABLES_, this->STITCH_REGEX_, this->variables_); // parse the vector line
+            //VectorParser::parseVectorLine(temp, line, this->STITCH_VARIABLES_, this->STITCH_REGEX_, this->variables_); // parse the vector line
+            VectorParser::parseVectorLine(temp, line, this->STITCH_VARIABLES_, this->STITCH_REGEX_VECTOR_, this->variables_); // parse the vector line
             this->valid_files_.push_back(temp); // add file mapping to matched files
         }
     }
     this->infile_.close();
 }
 
-string VectorPattern::inferPattern(const string& path, string& variables){
+string VectorPattern::inferPattern(const string& path, string& variables, const std::string& block_size){
```

Comment:
minor neat-picking - use `std` namespace explicitly

Reply:
We should also do a separate PR to revisit inferPattern and inferPatternInternal to make them const correct and see if we can do anything about not passing in non-const ref to `variables`.